File: 2021/0001_twoSum/0628/twoSum.ts
function twoSum(nums: number[], target: number): number[] {
    let map = new Map();
    let len = nums.length;
    for (let i = 0; i < len; i++) {
        if (map.has(target - nums[i])) {
            return [map.get(target - nums[i]), i];
        }
        map.set(nums[i], i);
    }
    return [];
}

File: 2021/0001_twoSum/0814/twoSum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        use std::collections::HashMap;

        let len = nums.len();
        let mut map = HashMap::with_capacity(len);
        for j in 0..len {
            if let Some(i) = map.get(&(target - nums[j])).copied() {
                return vec![i as i32, j as i32]
            } 
            map.insert(nums[j], j);
        }
        vec![]
    }
}
